ci-operator/step-registry/redhat-performance/benchmark-runner/03-workloads/redhat-performance-benchmark-runner-03-workloads-commands.sh

```
@@ -4,48 +4,83 @@ set -euo pipefail
   exit 1
 fi
 
+# SOCKS proxy for private VLAN clusters (API on 198.18.x.x)
+CLUSTER_SERVER=$(oc config view --kubeconfig=/tmp/kubeconfig --minify -o jsonpath='{.clusters[0].cluster.server}' 2>/dev/null || true)
+if [[ -n "${BASTION_ADDRESS:-}" ]]; then
+  if ! curl -sk --max-time 5 "${CLUSTER_SERVER}/version" &>/dev/null; then
+    echo "API not directly reachable, starting SOCKS proxy through bastion"
+    SOCKS_PORT=$((RANDOM % 55536 + 10000))
+    ES_PORT=$(<"/secret/elasticsearch_port")
+    ES_PORT="${ES_PORT%$'\n'}"
+    ssh ${SSH_ARGS} root@"${BASTION_ADDRESS}" -fNT -D "${SOCKS_PORT}" -L "127.0.0.1:${ES_PORT}:localhost:${ES_PORT}"
```

Shellcheck (0.11.0)

[info] Double quote to prevent globbing and word splitting.

(SC2086)